### ISSUE TITLE
[BUG] longhorn RWX doesn't work on freshly created volume

### ISSUE THREAD

**Issue by jstewart612** [longhorn/longhorn#6595]
## Describe the bug (🐛 if you encounter this issue)

I made a new Longhorn RWX volume.  The pods that reference it fail to start.

## To Reproduce

```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: rwx
  name: rwx
  namespace: wordpress
spec:
  replicas: 2
  selector:
    matchLabels:
      app: rwx
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: rwx
    spec:
      containers:
      - image: nginx:latest
        name: nginx
        resources: {}
        volumeMounts:
        - mountPath: /data
          name: data
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: rwx
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: rwx
  namespace: wordpress
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 10Gi
  storageClassName: longhorn-rwx
---
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: longhorn-rwx
parameters:
  fromBackup: ""
  migratable: "true"
  numberOfReplicas: "3"
  recurringJobSelector: '[{"name":"backup","task":"backup","cron":"5 0 * * *","retain":14}]'
  staleReplicaTimeout: "2880"
provisioner: driver.longhorn.io
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

## Expected behavior

My pods work with the rwx volume I want.

## Support bundle for troubleshooting

Too big for GitHub, but now here as of this edit: [support bundle](https://nextcloud.thecrimsontint.com/s/pnwYxTCxke9WP9b)

## Environment

 - Longhorn version: 1.5.1
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm through an ArgoCD Application
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s 1.27.3
   - Number of management node in the cluster: 5
   - Number of worker node in the cluster: 0
 - Node config
   - OS type and version: Almalinux 9.2
   - Kernel version: 5.14.0-284.25.1.el9_2.x86_64
   - CPU per node: 16
   - Memory per node: 64 GB
   - Disk type(e.g. SSD/NVMe/HDD): NVMe OS, SSD for Longhorn Data
   - Network bandwidth between the nodes: 2.5 Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 45

## Additional context

The ArgoCD application that installed longhorn:

```
---
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  annotations:
    argocd.argoproj.io/sync-wave: "1"
  finalizers:
    - resources-finalizer.argocd.argoproj.io
  name: longhorn
  namespace: argocd
spec:
  destination:
    namespace: longhorn-system
    server: 'https://kubernetes.default.svc'
  ignoreDifferences:
    - group: apps
      kind: Deployment
      name: longhorn-admission-webhook
    - group: apps
      kind: Deployment
      name: longhorn-conversion-webhook
    - group: apps
      kind: Deployment
      name: longhorn-recovery-backend
  project: default
  source:
    chart: longhorn
    helm:
      releaseName: longhorn
      values: |
        csi:
          kubeletRootDir: "/var/lib/kubelet"
        defaultSettings:
          backupTarget: "s3://longhorn@us-east-1/"
          backupTargetCredentialSecret: "longhorn-minio-creds"
          defaultDataLocality: "best-effort"
          replicaAutoBalance: "best-effort"
        longhornAdmissionWebhook:
          replicas: 0
        longhornConversionWebhook:
          replicas: 0
        longhornManager:
          log:
            format: json
        longhornRecoveryBackend:
          replicas: 0
        persistence:
          defaultDataLocality: "best-effort"
          defaultReplicaAutoBalance: "best-effort"
          migratable: true
          recurringJobSelector:
            enable: true
            jobList: '[{"name":"backup","task":"backup","cron":"5 0 * * *","retain": 14}]'
    repoURL: 'https://charts.longhorn.io'
    targetRevision: "*"
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
```

**Comment by derekbit** [comment]
Have you installed nfs-common (nfs-client) on each Longhorn node?

**Comment by derekbit** [comment]
for support bundle issue cc @c3y1huang

**Comment by jstewart612** [comment]
@derekbit in Almalinux 9, that's nfs-utils, and yes:

```
[root@nuc-1 ~]# dnf install nfs-utils
Last metadata expiration check: 2:24:27 ago on Sun 27 Aug 2023 05:33:38 AM EDT.
Package nfs-utils-1:2.5.4-18.el9.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
[root@nuc-1 ~]#
```

**Comment by derekbit** [comment]
Can you check the longhorn-manager pod log to see if there is any clue?

**Comment by jstewart612** [comment]
Out of manager:  nothing of note.

Out of longhorn-csi-plugin:

```
{"log":"time=\"2023-08-27T12:10:29Z\" level=error msg=\"NodeStageVolume: err: rpc error: code = InvalidArgument desc = volume pvc-37021370-3e83-4bcc-a9c2-a3ff2ae6b44e invalid controller count 2\"\n","stream":"stderr","time":"2023-08-27T12:10:29.383330371Z"}
```

**Comment by jstewart612** [comment]
If I try to view that volume in the web UI, I get a details page, but this pops up at the top:

```
failed to list snapshot: cannot get client for volume pvc-37021370-3e83-4bcc-a9c2-a3ff2ae6b44e: more than one engine exists
```

**Comment by c3y1huang** [comment]
> I'd love to but your UI also failed to do that. I followed the instructions, it sat at 20 percent and didn't move, and is still sitting there at 20 percent and not moving.
> 
> Pod logs so far:
> 
> ```
> [{"line":"{\"log\":\"+ exec tini -- /usr/bin/support-bundle-kit manager\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:18.245204832Z\"}","timestamp":"1693122620310873067","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=info msg=\\\"Running phase init\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.27196309Z\"}","timestamp":"1693122620310876344","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=debug msg=\\\"Create a local state store. (longhorn-system/support-bundle-2023-08-27t07-50-13z)\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.272493672Z\"}","timestamp":"1693122620310883238","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=debug msg=\\\"Get supportbundle longhorn-system/support-bundle-2023-08-27t07-50-13z state generating\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.27250706Z\"}","timestamp":"1693122620310891596","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=info msg=\\\"Succeed to run phase init. Progress (20).\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.272510491Z\"}","timestamp":"1693122620310892945","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=info msg=\\\"Running phase cluster bundle\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.272511913Z\"}","timestamp":"1693122620310897060","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=debug msg=\\\"Generating cluster bundle...\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.272513069Z\"}","timestamp":"1693122620310910462","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=debug msg=\\\"Get supportbundle longhorn-system/support-bundle-2023-08-27t07-50-13z\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.277439487Z\"}","timestamp":"1693122620310911972","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=info msg=\\\"Prepare to get all support bundle yamls!\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.277610216Z\"}","timestamp":"1693122620310918674","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:18Z\\\" level=info msg=\\\"[Cluster] generate YAMLs, yamlsDir: /tmp/support-bundle-kit/bundle/yamls\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:18.277617993Z\"}","timestamp":"1693122620310927076","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"W0827 07:50:18.309811       7 warnings.go:70] v1 ComponentStatus is deprecated in v1.19+\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:18.309888213Z\"}","timestamp":"1693122620310950507","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"time=\\\"2023-08-27T07:50:19Z\\\" level=info msg=\\\"[Default] generate YAMLs, yamlsDir: /tmp/support-bundle-kit/bundle/yamls\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2023-08-27T07:50:19.398751893Z\"}","timestamp":"1693122620310978394","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"W0827 07:50:19.776752       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:19.776814728Z\"}","timestamp":"1693122620310979181","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"W0827 07:50:20.416430       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:20.416531728Z\"}","timestamp":"1693122620561535415","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"W0827 07:50:20.787837       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:20.787889949Z\"}","timestamp":"1693122620812237610","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}},{"line":"{\"log\":\"W0827 07:50:21.018007       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:21.018071916Z\"}","timestamp":"1693122621062816892","fields":{"app":"support-bundle-manager","container":"manager","filename":"/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log","job":"longhorn-system/support-bundle-manager","namespace":"longhorn-system","node_name":"nuc-4","pod":"longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"}}]
> ```

Based on the log,  it looks like the support bundle manager is hanging on the YAML collection. There are also some warning logs related to `metallb.io v1beta1 AddressPool`, but it doesn't look like a blocker.
```
{
	"line": "{\"log\":\"W0827 07:50:19.776752       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:19.776814728Z\"}",
	"timestamp": "1693122620310979181",
	"fields": {
		"app": "support-bundle-manager",
		"container": "manager",
		"filename": "/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log",
		"job": "longhorn-system/support-bundle-manager",
		"namespace": "longhorn-system",
		"node_name": "nuc-4",
		"pod": "longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"
	}
}, {
	"line": "{\"log\":\"W0827 07:50:20.416430       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:20.416531728Z\"}",
	"timestamp": "1693122620561535415",
	"fields": {
		"app": "support-bundle-manager",
		"container": "manager",
		"filename": "/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log",
		"job": "longhorn-system/support-bundle-manager",
		"namespace": "longhorn-system",
		"node_name": "nuc-4",
		"pod": "longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"
	}
}, {
	"line": "{\"log\":\"W0827 07:50:20.787837       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:20.787889949Z\"}",
	"timestamp": "1693122620812237610",
	"fields": {
		"app": "support-bundle-manager",
		"container": "manager",
		"filename": "/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log",
		"job": "longhorn-system/support-bundle-manager",
		"namespace": "longhorn-system",
		"node_name": "nuc-4",
		"pod": "longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"
	}
}, {
	"line": "{\"log\":\"W0827 07:50:21.018007       7 warnings.go:70] metallb.io v1beta1 AddressPool is deprecated, consider using IPAddressPool\\n\",\"stream\":\"stderr\",\"time\":\"2023-08-27T07:50:21.018071916Z\"}",
	"timestamp": "1693122621062816892",
	"fields": {
		"app": "support-bundle-manager",
		"container": "manager",
		"filename": "/var/log/pods/longhorn-system_longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7_836a5c21-9101-4d2d-8c3a-8c01913d10e5/manager/0.log",
		"job": "longhorn-system/support-bundle-manager",
		"namespace": "longhorn-system",
		"node_name": "nuc-4",
		"pod": "longhorn-support-bundle-manager-support-bundle-2023-08-27tmqms7"
	}
}]
```
@jstewart612, 
1. What is the output of `kubectl -n longhorn-system get supportbundles`?
1. Could you please share the logs for the `longhorn-support-bundle-manager-support-bundle-*`?

**Comment by jstewart612** [comment]
kubectl -n longhorn-system get supportbundles

```
NAME                                  STATE              ISSUE                                              DESCRIPTION                                                     AGE
support-bundle-2023-08-27t07-50-13z   ReadyForDownload   https://github.com/longhorn/longhorn/issues/6595   Support bundle for my Longhorn install for Github issue #6595   20h
```

Now, if you read above, at the time of the comment, I could not, because the window sat at 20 percent and I could not do anything.  The status also was not ReadyForDownload.  So, at the time, this was a silly ask, because I could not comply.

Checking anew now, though, it seems it finally made one.  A nearly 1 GB file, from the looks.  Guessing it just took a real long time.

So... Github only allows files up to 25 MB and this is 991 MB.  I have to upload to alternate location.  Uploaded here: https://nextcloud.thecrimsontint.com/s/pnwYxTCxke9WP9b

**Comment by jstewart612** [comment]
Interesting development... so I updated replicas to 1 on the volume and replicas 1 on the deployment.  Now, one pod and only one pod can mount the volume at a time.

Even more interestingly, inside the container that can mount it:

```
root@rwx-7fc7ccf7d8-qjgh8:/# df -h
Filesystem                                              Size  Used Avail Use% Mounted on
overlay                                                 829G   28G  801G   4% /
tmpfs                                                    64M     0   64M   0% /dev
/dev/longhorn/pvc-37021370-3e83-4bcc-a9c2-a3ff2ae6b44e  9.8G   24K  9.8G   1% /data
/dev/mapper/almalinux-root                              829G   28G  801G   4% /etc/hosts
shm                                                      64M     0   64M   0% /dev/shm
tmpfs                                                    63G   12K   63G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                                    32G     0   32G   0% /proc/asound
tmpfs                                                    32G     0   32G   0% /proc/acpi
tmpfs                                                    32G     0   32G   0% /proc/scsi
tmpfs                                                    32G     0   32G   0% /sys/firmware
root@rwx-7fc7ccf7d8-qjgh8:/#
```

iirc, your rwx implementation makes an nfs server, and that's what I should expect to see here... but I don't.  That looks like what you all do for RWO and not RWX.  Is the ReadWriteMany accessMode somehow being ignored and it's treating this like ReadWriteOnce for some reason?

**Comment by jstewart612** [comment]
Any luck?

**Comment by ChanYiLin** [comment]
Hi @jstewart612 
Can you provide the details of the pod that couldn't start by `kubectl get pod/XXX -oyaml -n XXX`?
I would like to see the error message in the pod event, it might have the information why the volume failed to mount to the pod.

**Comment by ChanYiLin** [comment]
Hi @jstewart612 
To achieve rwx, you don't need to create another StorageClass for that
you just need to specify `ReadWriteMany` in the PVC or during Volume creation using UI
```
  accessModes:
    - ReadWriteMany
```
Then when creating pods with the PVC, Longhorn will automatically create share-manager(NFS) to mount to multiple pods.

What happened in your case is that your storageclass has parameter `migratable: "true"`
which is used for volume migration in Harvester (VM migration).
In Volume migration scenario, we also need to create a PVC with `ReadWriteMany` and StorageClass with `migratable: "true"`, but this time longhorn wouldn't create share-manager and the PVC can only mount to one pod (attach to one node).

**Comment by ChanYiLin** [comment]
You probably refer to the examples here in this folder https://github.com/longhorn/longhorn/tree/master/examples/rwx
But you just need to use the `rwx-nginx-deployment.yaml`
We will move `storageclass-migratable.yaml` out of the folder for it is indeed quite confused

**Comment by rock1013** [comment]
Hi, I got the same problem. the pod flapping  pend,create,run,error,terminiate. I managed to catch the nfs pod log like this. `<html>
<body>


 
--
time="2023-09-04T10:02:28Z" level=info msg="starting RLIMIT_NOFILE rlimit.Cur 1048576, rlimit.Max 1048576"
time="2023-09-04T10:02:28Z" level=info msg="Ending RLIMIT_NOFILE rlimit.Cur 1048576, rlimit.Max 1048576"
time="2023-09-04T10:02:28Z" level=info msg="Listening on share manager gRPC server :9600"
time="2023-09-04T10:02:28Z" level=debug msg="Volume pvc-011cb722-d477-4e78-919f-cabc5133e338 device /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 contains filesystem of format ext4" encrypted=false volume=pvc-011cb722-d477-4e78-919f-cabc5133e338
I0904 10:02:28.431855 1 mount_linux.go:389] `fsck` error fsck from util-linux 2.37.2
/dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 is in use.
e2fsck: Cannot continue, aborting.
 
 
E0904 10:02:28.435104 1 mount_linux.go:195] Mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 /export/pvc-011cb722-d477-4e78-919f-cabc5133e338
Output: mount: /export/pvc-011cb722-d477-4e78-919f-cabc5133e338: /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 already mounted or mount point busy.
 
time="2023-09-04T10:02:28Z" level=warning msg="Failed to mount volume" encrypted=false error="mount failed: exit status 32\nMounting command: mount\nMounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 /export/pvc-011cb722-d477-4e78-919f-cabc5133e338\nOutput: mount: /export/pvc-011cb722-d477-4e78-919f-cabc5133e338: /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 already mounted or mount point busy.\n" volume=pvc-011cb722-d477-4e78-919f-cabc5133e338
W0904 10:02:28.487130 1 mount_helper_common.go:133] Warning: "/export/pvc-011cb722-d477-4e78-919f-cabc5133e338" is not a mountpoint, deleting
time="2023-09-04T10:02:28Z" level=debug msg="Device /dev/mapper/pvc-011cb722-d477-4e78-919f-cabc5133e338 is not an active LUKS device" error="failed to run cryptsetup args: [status pvc-011cb722-d477-4e78-919f-cabc5133e338] output: error: exit status 4"
time="2023-09-04T10:02:28Z" level=fatal msg="Error running start command: mount failed: exit status 32\nMounting command: mount\nMounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 /export/pvc-011cb722-d477-4e78-919f-cabc5133e338\nOutput: mount: /export/pvc-011cb722-d477-4e78-919f-cabc5133e338: /dev/longhorn/pvc-011cb722-d477-4e78-919f-cabc5133e338 already mounted or mount point busy.\n."


</body>
</html>`
